org/qmckl_sherman_morrison_woodbury.cpp

```diff
@@ -0,0 +1,535 @@
+    for (unsigned int j = 0; j < 3; j++) {
+      C[i * 3 + j] = 0;
+      for (unsigned int k = 0; k < Dim; k++) {
+        C[i * 3 + j] += Slater_inv[i * Dim + k] * Updates[Dim * j + k];
+      }
+    }
+  }
+
+#ifdef DEBUG2
+  showMatrix2(C, Dim, 3, "C = S_inv * U");
+  showMatrix2(D, 3, Dim, "D = V * S_inv");
+#endif
+
+  // Compute B = 1 + V.C
+  const double B0 = C[row1 * 3] + 1;
+  const double B1 = C[row1 * 3 + 1];
+  const double B2 = C[row1 * 3 + 2];
+  const double B3 = C[row2 * 3];
+  const double B4 = C[row2 * 3 + 1] + 1;
+  const double B5 = C[row2 * 3 + 2];
+  const double B6 = C[row3 * 3];
+  const double B7 = C[row3 * 3 + 1];
+  const double B8 = C[row3 * 3 + 2] + 1;
+
+#ifdef DEBUG2
+  showMatrix2(B, 3, 3, "B = 1 + V * C");
+#endif
+
+  // Check if determinant of B is not too close to zero
+  double det;
+  det = B0 * (B4 * B8 - B5 * B7) - B1 * (B3 * B8 - B5 * B6) +
+        B2 * (B3 * B7 - B4 * B6);
+#ifdef DEBUG2
+  std::cerr << "Determinant of B = " << det << std::endl;
+#endif
+  if (std::fabs(det) < threshold()) {
+#ifdef DEBUG1
+    std::cerr << "Determinant too close to zero! No inverse found."
+              << std::endl;
+    std::cerr << "Determinant = " << det << std::endl;
+#endif
+    return false;
+  }
+
+  // Compute B^{-1} with explicit formula for 3x3 inversion
+  double Binv[9], idet = 1.0 / det;
+  Binv[0] = (B4 * B8 - B7 * B5) * idet;
+  Binv[1] = -(B1 * B8 - B7 * B2) * idet;
+  Binv[2] = (B1 * B5 - B4 * B2) * idet;
+  Binv[3] = -(B3 * B8 - B6 * B5) * idet;
+  Binv[4] = (B0 * B8 - B6 * B2) * idet;
+  Binv[5] = -(B0 * B5 - B3 * B2) * idet;
+  Binv[6] = (B3 * B7 - B6 * B4) * idet;
+  Binv[7] = -(B0 * B7 - B6 * B1) * idet;
+  Binv[8] = (B0 * B4 - B3 * B1) * idet;
+
+#ifdef DEBUG2
+  std::cerr << "Conditioning number of B = " << condition1(B, Binv, 3)
+            << std::endl;
+  showMatrix2(Binv, 3, 3, "Binv");
+#endif
+
+  // Compute tmp = B^{-1} x (V.S^{-1})
+  double tmp[3 * Dim];
+  for (unsigned int i = 0; i < 3; i++) {
+    for (unsigned int j = 0; j < Dim; j++) {
+      tmp[i * Dim + j] = Binv[i * 3] * Slater_inv[row1 * Dim + j];
+      tmp[i * Dim + j] += Binv[i * 3 + 1] * Slater_inv[row2 * Dim + j];
+      tmp[i * Dim + j] += Binv[i * 3 + 2] * Slater_inv[row3 * Dim + j];
+    }
+  }
+
+#ifdef DEBUG2
+  showMatrix2(tmp, 3, Dim, "tmp = Binv * D");
+#endif
+
+  // Compute (S + U V)^{-1} = S^{-1} - C x tmp
+  for (unsigned int i = 0; i < Dim; i++) {
+    for (unsigned int j = 0; j < Dim; j++) {
+      Slater_inv[i * Dim + j] -= C[i * 3] * tmp[j];
+      Slater_inv[i * Dim + j] -= C[i * 3 + 1] * tmp[Dim + j];
+      Slater_inv[i * Dim + j] -= C[i * 3 + 2] * tmp[2 * Dim + j];
+    }
+  }
+
+#ifdef DEBUG2
+  showMatrix2(Slater_inv, Dim, Dim, "Slater_inv AFTER update");
+#endif
+  return true;
+}
+
+// Sherman Morrison, with J. Slagel splitting (caller function)
+// http://hdl.handle.net/10919/52966
+void sherman_morrison_splitting_caller(double *Slater_inv, unsigned int Dim, unsigned int N_updates,
```

Comment:
Swap the names between `sherman_morrison_splitting_caller` and `sherman_morrisson_splitting`.

Maybe call the internal `slagel_splitting` ?

Declare the internal one with `static` keyword.

Reply:
Done.